modules/meshrenderinggl/glsl/oit/sort.glsl

```diff
@@ -105,27 +105,33 @@ void fillFragmentArraySortedUntilDepth(uint pixelIdx, float maxDepth, out int nu
     // Load fragments into a local memory array for sorting
     numFrag = 0;
     float depth = 0.0;
+    uint lastPtr = 0;
 
     while (numFrag < ABUFFER_SIZE) {
-        vec4 val = selectionSortNext(pixelIdx, depth);
+        vec4 val = selectionSortNext(pixelIdx, depth, lastPtr);
         depth = val.y;
         if (depth < 0.0 || depth > maxDepth) return;
         fragmentList[numFrag] = val;
         numFrag++;
     }
 }
 
-vec4 selectionSortNext(uint headPtr, float lastDepth) {
+vec4 selectionSortNext(uint headPtr, float lastDepth, inout uint lastPtr) {
     vec4 closestVal = vec4(-1.0);
     float minDepth = 1.0;
+    bool passedLastPtr = false;
+    uint minPtr = 0;
```

Comment:
Why not headPtr?

Reply:
We need to keep the pointer to the element we return for correct ordering later on. See other comments.